### ISSUE TITLE
Instant picture upload subfolder naming

### ISSUE THREAD

**Issue by MHBraun** [nextcloud/android#1330]
### Actual behaviour
-Tell us what happens

Naming convention:
yyyy = Year in digit
mm = Month in digit
dd = Day of the month in digit
Condition: 
Settings - Use subfolders - yes

With uploading pictures to the choosen upload folder a subfolder yyyy is generated and a subfolder thereof  mm and the picture file is put into this folder. Which results in
InstantUpload/yyyy/mm/filename (= InstantUpload/2019/08/20180801-140303.jpg)
This does create multiple mm folders (same name) with different content in different years. 

### Expected behaviour
-Tell us what should happen

 The naming of the subfolder(s) should be configurable with a selectable given pattern.
Then according to the need of the user a pattern like
InstantUpload/yyyy/yyyymm/ (= InstantUpload/2018/201808/)
or
InstantUpload/yyyy/yyyymm/yyyymmdd/ (=InstantUpload/2018/201808/20180801/)
would be possible.

This would integrate with other picture management software.

### Steps to reproduce
1. Setting - subfolders - yes
2. Take a picture
3. Verify the folder structure in InstantUpload

### Environment data
Android version: 4.4.2

Device model: SM-G900FD

Stock or customized system: Samsung stock

Nextcloud app version: 1.4.3

Nextcloud server version: 12.0.1

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by tobiasKaminsky** [comment]
While I understand the problem I am not sure if this is the best way to solve the problem nor I am sure that this is UI/UX wise the best as it adds more configurations.

Either we find a more generic approach to configure this or I vote against adding "dd" (days) as subfolders.

@nextcloud/designers

**Comment by jancborchardt** [comment]
Yeah, agree with you @tobiasKaminsky. Adding configuration options here will also just increase the testing matrix crazily.

**Comment by MHBraun** [comment]
Understand.
Then please change the subfolder mm to yyyymm. 
This is way less flexible for the user but it solves the issue.
This is probably the generic approach you are looking for. :-)

**Comment by jd1** [comment]
I'm missing this feature too. I configured my old backup up to move the files to a folder with the name yyyy-mm-dd. Additionally I added folder for the current year. So my images are in a structure like this: yyyy/yyyy-mm-dd/

When I switched to NC I can only choose yyyy/mm/ so all files of one month are in the same folder. Since I move files from different sources(Camera/Messengers/Screenshots) to the same backup folder, they are mixed up because they follow different naming schemas. I would really appreciate it to get at least a folder for every day.

A generic approach would be to let the user define the pattern himself, but I think that's not very user-friendly. Maybe you could add this as an "expert" setting and the normal user gets the choice between a few patterns.

Here is a screenshot of the app I'm using right now. In my opinion, these are too many choices, but it's just an inspiration :) 
![Screenshot](https://jd.isobeef.org/img/Screenshot_20180322-101150.jpg)

**Comment by MHBraun** [comment]
The workaround is using EXIFTOOLS on the serverside to copy and rename the files into a different directory. Not really clever, as the files are doubled, but at least all pictures from all sources in the same structure.
Still waiting for the feature :)

**Comment by joshtrichards** [comment]
FYI: Subfolder options were expanded in #11852

**Comment by joshtrichards** [comment]
Related: #359
